nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/storage/FetchAzureDataLakeStorage.java

```diff
@@ -67,6 +67,10 @@ public void onTrigger(ProcessContext context, ProcessSession session) throws Pro
             final DataLakeDirectoryClient directoryClient = dataLakeFileSystemClient.getDirectoryClient(directory);
             final DataLakeFileClient fileClient = directoryClient.getFileClient(fileName);
 
+            if (fileClient.getProperties().isDirectory()) {
+                throw new ProcessException(FILE.getDisplayName() + " (" + fileName + ") points to a directory. Full path: " + fileClient.getFilePath());
```

Comment:
The Exception I am getting is FetchAzureDataLakeStorage[id=xxxxx] failed to process session due to com.azure.storage.file.datalake.models.PathProperties.isDirectory()Ljava/lang/Boolean;; Processor Administratively Yielded for 1 sec: java.lang.NoSuchMethodError: com.azure.storage.file.datalake.models.PathProperties.isDirectory()Ljava/lang/Boolean;

Reply:
It looks like switching to version 12.1.1 of azure-storage-file-datalake instead of 12.0.1 did not take effect in your build. Can you please verify that you rebuilt the entire nifi-azure-bundle and used the newly generated nifi-azure-nar in your running instance of NiFi (after restart)?

Reply:
It works properly for me too. I also updated the nifi-azure-services-api-nar (copy to NiFi's lib folder). Though, it might not necessary, only the nifi-azure-nar.